src/main/java/io/snyk/dylansnyk/demo/controller/EchoController.java

```diff
@@ -33,4 +35,19 @@ public EchoDto getEcho(@RequestParam String echo) {
     public void executeCommand(@NonNull final String[] command) throws IOException {
         Process process = new ProcessBuilder(command).start();
     }
+    
+    @SneakyThrows
+    @GetMapping("/ssrf")
+    public void makeRequest(@RequestParam String url) {
+        HttpClient client = HttpClient.newHttpClient();
+
+        HttpRequest request = HttpRequest.newBuilder()
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; Server-Side Request Forgery (SSRF)

Unsanitized input from an HTTP parameter flows into uri, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.

<h6>
<a href="https://cwe.mitre.org/data/definitions/918.html">CWE-918</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 834</a> | Line 44</h6>